Services/Statistics/SimulationStatistics.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Clustering;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Statistics
+{
+    public interface ISimulationStatistics
+    {
+        Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string simulationId);
+        Task CreateOrUpdateAsync(string simulationId, SimulationStatisticsModel statistics);
+    }
+
+    public class SimulationStatistics : ISimulationStatistics
+    {
+        private readonly IStorageRecords simulationStatisticsStorage;
+        private readonly IClusterNodes clusterNodes;
+        private readonly ISimulations simulations;
+        private readonly IServicesConfig config;
+        private readonly ILogger log;
+
+        public SimulationStatistics(IServicesConfig config,
+            IClusterNodes clusterNodes,
+            ISimulations simulations,
+            IFactory factory,
+            ILogger logger)
+        {
+            this.clusterNodes = clusterNodes;
+            this.simulations = simulations;
+            this.config = config;
+            this.log = logger;
+            this.simulationStatisticsStorage = factory.Resolve<IStorageRecords>().Init(config.StatisticsStorage);
+        }
+
+        public async Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string simulationId)
+        {
+            if (string.IsNullOrEmpty(simulationId))
+            {
+                this.log.Error("Simulation Id cannot be null or empty");
+                return null;
+            }
+
+            string sqlCondition = " CONTAINS(ROOT.id, @simulationId)";
+            SqlParameter[] sqlParameters = new[] { new SqlParameter { Name = "@simulationId", Value = simulationId } };
```

Comment:
this approach couples the solution with Cosmos DB, which is going to be a problem if we want to change storage type, e.g. a customer in RM or partners want to store data differently

Reply:
Existing implementation (StorageRecords.cs) had cosmosdb specific implementation. I wanted to pass sql query to Get/Delete operation for all simulation stats records. If we want to avoid using SqlParameter and take a dependency on Microsoft.Azure.Documents, I can change query to get/delete one record at a time (i.e call StorageRecord.GetAsync(string id))